### ISSUE TITLE
Underlying socket connection gone?

### ISSUE THREAD

**Issue by Nirusu** [neo4j-contrib/neomodel#474]
I'm not sure if I really understand the usage here, but I have the following script:

```
from neomodel import (config, StructuredNode, StringProperty, IntegerProperty,
    UniqueIdProperty, RelationshipTo)

config.DATABASE_URL = 'bolt://neo4j:badpassword@localhost:7687'

class Person(StructuredNode):
    uid = UniqueIdProperty()
    name = StringProperty(unique_index=True)
    age = IntegerProperty(index=True, default= 0)


def test():
    list= ["Jim", "Hans", "Rainer"]
    for name_from_list in list:
        node = Person(name = name_from_list).save() 
        node.name = name_from_list
        node.save()

test()
```

When running this script, the entries are added, but the scripts exits with the following error:

```
Failed to write data to connection Address(host='localhost', port=7687) (Address(host='127.0.0.1', port=7687)); ("0; 'Underlying socket connection gone (_ssl.c:2084)'")
Failed to write data to connection Address(host='localhost', port=7687) (Address(host='127.0.0.1', port=7687)); ("0; 'Underlying socket connection gone (_ssl.c:2084)'")
```

Now, according to Google this happens if you don't close the connection from the driver, but I though by using neomodel, I wouldn't have to do this?

Setup is Ubuntu 18.04 with Python 3.6, Neo4j from Docker Hub and neomodel v.3.3.2 installed via pip.

**Comment by SkullCarverCoder** [comment]
I'm also having this issue since the new version 3.3.2. Will be watching and see if I can find a fix

**Comment by laurentS** [comment]
I think the issue is related to [this bug](https://github.com/neo4j/neo4j-python-driver/issues/293) in `neo4j-driver`.

**Comment by SkullCarverCoder** [comment]
I just solved this problem and subsequent ones with the following:

`pip install --force-reinstall neo4j==1.7.2 neobolt==1.7.9 neotime==1.7.4`

and change in your `config.DATABASE_URL` to "127.0.0.1" instead of "localhost" in the URL (same if you use  `neomodel.db.set_connection` somewhere), after that I ran with no problem and neomodel shouldn't give you any trouble about connection! 😉

**Comment by Nirusu** [comment]
Yeah, going back to these versions seems to work. However, what's even more interesting was that I tried experimenting a bit with the versions of neo4j and neobolt and while previously I had issues with the most recent versions, after switching around a couple of times it worked fine... Until I uninstalled everything and reinstalled everything again on the most recent version.

I described this in detail in the other issue. As a workaround, downgrading either the server or the packages seem to work, but of course this cannot be the solution. But as it looks like, this is not a specific issue of neomodel but rather of the packages it uses.

**Comment by aanastasiou** [comment]
@Nirusu Thanks. neomodel [only specifies the driver version](https://github.com/neo4j-contrib/neomodel/blob/master/setup.py#L20). I will leave this issue open for some time in case it is of help to others. It looks like something that needs to be addressed at the level of the driver's dependencies.

**Comment by mariusconjeaud** [comment]
Probably fixed by many driver updates since then. Closing this for cleaning up the Issues log, feel free to re-open if relevant.
